### PR TITLE
Integrate CustomerSession directly into CustomerSheet

### PR DESCRIPTION
**Pull request by wooj-stripe**
## Summary
Integrate CustomerSession directly into CustomerSheet

## Motivation
Change in direction of integration shape

## Testing
Relying on existing UI tests for CustomerSession + Customer Sheet

### CODE REVIEW

**Comment by oleksandr-vonahi** [comment]
This is massive. @gooroodev can you help reviewing it?

**Comment by admsev** [comment]
This is massive. @gooroodev would you mind reviewing it?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheetDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        }
+    }
+
+    func fetchSavedPaymentMethods() async throws -> [STPPaymentMethod]? {
+        switch dataSource {
+        case .customerAdapter(let customerAdapter):
+            return try await customerAdapter.fetchPaymentMethods()
+        case .customerSession(let customerSessionAdapter):
+            let elementsSessionResult = try await customerSessionAdapter.elementsSession()
+            return elementsSessionResult.customer?.paymentMethods ?? []
```
Comment (davidme-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheetDataSource.swift:124]: Should this return nil or an empty array?

**Review by davidme-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheetDataSource.swift (hunk 1 above)
Comment (wooj-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheetDataSource.swift]: Oof, great catch.  In the event that we fail to get payment methods, we need to return nil, which corresponds to dismissing the sheet and force users to effectively re-open the sheet.

**Review by davidme-stripe — APPROVED** [review]

**Pull request merged**